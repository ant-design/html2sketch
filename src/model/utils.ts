import FileFormat from '@sketch-hq/sketch-file-format-ts';

export const defaultExportOptions: FileFormat.ExportOptions = {
  _class: 'exportOptions',
  includedLayerIds: [],
  layerOptions: 0,
  shouldTrim: false,
  exportFormats: [],
};

/**
 * SKetch默认的圆角选项
 * */
export const defaultBorderOptions: FileFormat.BorderOptions = {
  _class: 'borderOptions',
  isEnabled: true,
  dashPattern: [],
  lineCapStyle: FileFormat.LineCapStyle.Butt,
  lineJoinStyle: FileFormat.LineJoinStyle.Miter,
};

/**
 * SKetch默认的色彩控制
 * */
export const defaultColorControls: FileFormat.ColorControls = {
  _class: 'colorControls',
  isEnabled: false,
  brightness: 0,
  contrast: 1,
  hue: 0,
  saturation: 1,
};

/**
 * SKetch 默认规则数据
 * */
export const defaultRuleData = (): FileFormat.RulerData => ({
  _class: 'rulerData',
  base: 0,
  guides: [],
});

/**
 * SKetch 默认blend 样式
 * */
export const defaultContextSettings: FileFormat.GraphicsContextSettings = {
  _class: 'graphicsContextSettings',
  blendMode: FileFormat.BlendMode.Normal,
  opacity: 1,
};

export const defaultGradient: FileFormat.Gradient = {
  _class: 'gradient',
  elipseLength: 0,
  from: '0.5 0',
  to: '0.5 0',
  stops: [],
  gradientType: FileFormat.GradientType.Linear,
};

export const defaultNodeStyle: Partial<CSSStyleDeclaration> = {
  backgroundColor: 'rgba(0, 0, 0, 0)',
  backgroundImage: 'none',
  borderWidth: '0px',
  boxShadow: 'none',
  // verticalAlign: 'baseline',
};
