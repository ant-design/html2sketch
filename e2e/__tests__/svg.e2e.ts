import { initHtml2Sketch, isUpdate, outputJSONData } from './utils';
import SvgIcon from './json/svg-icon.json';

describe('测试 Svg icon 类型', () => {
  test('生成符合预期的规范', async () => {
    const html2Sketch = await initHtml2Sketch();
    function selector(dom: Document) {
      return dom.getElementById('icons')!;
    }

    const symbol = await html2Sketch.nodeToSketchSymbol('/basic/svg', selector);
    expect(symbol).toBeTruthy();
    expect(symbol).toMatchSnapshot();

    if (isUpdate) {
      outputJSONData(symbol, 'svg-icon');
    }
    expect(symbol).toStrictEqual(SvgIcon);
  });
});
