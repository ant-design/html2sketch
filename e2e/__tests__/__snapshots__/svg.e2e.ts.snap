// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 Svg icon 类型 生成符合预期的规范 1`] = `
Object {
  "_class": "symbolMaster",
  "allowsOverrides": true,
  "backgroundColor": Object {
    "_class": "color",
    "alpha": 1,
    "blue": 1,
    "green": 1,
    "red": 1,
  },
  "booleanOperation": -1,
  "changeIdentifier": 0,
  "do_objectID": "UUID",
  "exportOptions": Object {
    "_class": "exportOptions",
    "exportFormats": Array [],
    "includedLayerIds": Array [],
    "layerOptions": 0,
    "shouldTrim": false,
  },
  "frame": Object {
    "_class": "rect",
    "constrainProportions": false,
    "height": 22,
    "width": 238,
    "x": 343,
    "y": 150,
  },
  "groupLayout": Object {
    "_class": "MSImmutableFreeformGroupLayout",
  },
  "hasBackgroundColor": false,
  "hasClickThrough": true,
  "horizontalRulerData": Object {
    "_class": "rulerData",
    "base": 0,
    "guides": Array [],
  },
  "includeBackgroundColorInExport": true,
  "includeBackgroundColorInInstance": false,
  "includeInCloudUpload": true,
  "isFixedToViewport": false,
  "isFlippedHorizontal": false,
  "isFlippedVertical": false,
  "isFlowHome": false,
  "isLocked": false,
  "isVisible": true,
  "layerListExpandedType": 0,
  "layers": Array [
    Object {
      "_class": "group",
      "booleanOperation": -1,
      "clippingMaskMode": 0,
      "do_objectID": "UUID",
      "exportOptions": Object {
        "_class": "exportOptions",
        "exportFormats": Array [],
        "includedLayerIds": Array [],
        "layerOptions": 0,
        "shouldTrim": false,
      },
      "frame": Object {
        "_class": "rect",
        "constrainProportions": false,
        "height": 14,
        "width": 14,
        "x": 0,
        "y": 3.75,
      },
      "groupLayout": Object {
        "_class": "MSImmutableFreeformGroupLayout",
      },
      "hasClickThrough": false,
      "hasClippingMask": false,
      "isFixedToViewport": false,
      "isFlippedHorizontal": false,
      "isFlippedVertical": false,
      "isLocked": false,
      "isVisible": true,
      "layerListExpandedType": 0,
      "layers": Array [
        Object {
          "_class": "shapePath",
          "booleanOperation": -1,
          "clippingMaskMode": 0,
          "do_objectID": "UUID",
          "edited": true,
          "exportOptions": Object {
            "_class": "exportOptions",
            "exportFormats": Array [],
            "includedLayerIds": Array [],
            "layerOptions": 0,
            "shouldTrim": false,
          },
          "frame": Object {
            "_class": "rect",
            "constrainProportions": false,
            "height": 4.09378781225861,
            "width": 5.81133329278406,
            "x": 4.09511460360797,
            "y": 4.57808718774139,
          },
          "isClosed": true,
          "isFixedToViewport": false,
          "isFlippedHorizontal": false,
          "isFlippedVertical": false,
          "isLocked": false,
          "isVisible": true,
          "layerListExpandedType": 0,
          "name": "路径",
          "nameIsFixed": false,
          "pointRadiusBehaviour": 1,
          "points": Array [
            Object {
              "_class": "curvePoint",
              "cornerRadius": 0,
              "curveFrom": "{0.5133248399992646, 0.004690048149804383}",
              "curveMode": 4,
              "curveTo": "{0.5173421906682137, 0.012604540006713702}",
              "hasCurveFrom": true,
              "hasCurveTo": false,
              "point": "{0.5173421906682137, 0.012604540006713702}",
            },
            Object {
              "_class": "curvePoint",
              "cornerRadius": 0,
              "curveFrom": "{0.493128049928784, -1.3559850626608662e-17}",
              "curveMode": 3,
              "curveTo": "{0.5068719500712158, -1.3559850626608662e-17}",
              "hasCurveFrom": true,
              "hasCurveTo": true,
              "point": "{0.49999999999999994, -1.3559850626608662e-17}",
            },
            Object {
              "_class": "curvePoint",
              "cornerRadius": 0,
              "curveFrom": "{0.48265780933178615, 0.012604540006713719}",
              "curveMode": 4,
              "curveTo": "{0.48667516000073524, 0.004690048149804397}",
              "hasCurveFrom": false,
              "hasCurveTo": true,
              "point": "{0.48265780933178615, 0.012604540006713719}",
            },
            Object {
              "_class": "curvePoint",
              "cornerRadius": 0,
              "curveFrom": "{-0.006149983456007961, 0.9717559860689973}",
              "curveMode": 4,
              "curveTo": "{0.004067121123714938, 0.9515271652805768}",
              "hasCurveFrom": true,
              "hasCurveTo": false,
              "point": "{0.004067121123714938, 0.9515271652805768}",
            },
            Object {
              "_class": "curvePoint",
              "cornerRadius": 0,
              "curveFrom": "{0.021543747378504056, 1}",
              "curveMode": 4,
              "curveTo": "{0.004067121123714938, 1}",
              "hasCurveFrom": false,
              "hasCurveTo": true,
              "point": "{0.021543747378504056, 1}",
            },
            Object {
              "_class": "curvePoint",
              "cornerRadius": 0,
              "curveFrom": "{0.1750691872474976, 1}",
              "curveMode": 4,
              "curveTo": "{0.14764432758613624, 1}",
              "hasCurveFrom": true,
              "hasCurveTo": false,
              "point": "{0.14764432758613624, 1}",
            },
            Object {
              "_class": "curvePoint",
              "cornerRadius": 0,
              "curveFrom": "{0.21728196143214204, 0.9496187859609144}",
              "curveMode": 4,
              "curveTo": "{0.20114969104310593, 0.9812978826673093}",
              "hasCurveFrom": false,
              "hasCurveTo": true,
              "point": "{0.21728196143214204, 0.9496187859609144}",
            },
            Object {
              "_class": "curvePoint",
              "cornerRadius": 0,
              "curveFrom": "{0.4998655644134246, 0.3946620798031108}",
              "curveMode": 1,
              "curveTo": "{0.4998655644134246, 0.3946620798031108}",
              "hasCurveFrom": false,
              "hasCurveTo": false,
              "point": "{0.4998655644134246, 0.3946620798031108}",
            },
            Object {
              "_class": "curvePoint",
              "cornerRadius": 0,
              "curveFrom": "{0.7988503089568939, 0.9812978826673089}",
              "curveMode": 4,
              "curveTo": "{0.7827180385678578, 0.9496187859609144}",
              "hasCurveFrom": true,
              "hasCurveTo": false,
              "point": "{0.7827180385678578, 0.9496187859609144}",
            },
            Object {
              "_class": "curvePoint",
              "cornerRadius": 0,
              "curveFrom": "{0.8523556724138636, 1}",
              "curveMode": 4,
              "curveTo": "{0.8246619415793518, 1}",
              "hasCurveFrom": false,
              "hasCurveTo": true,
              "point": "{0.8523556724138636, 1}",
            },
            Object {
              "_class": "curvePoint",
              "cornerRadius": 0,
              "curveFrom": "{0.9959328788762849, 1}",
              "curveMode": 4,
              "curveTo": "{0.9784562526214958, 1}",
              "hasCurveFrom": true,
              "hasCurveTo": false,
              "point": "{0.9784562526214958, 1}",
            },
            Object {
              "_class": "curvePoint",
              "cornerRadius": 0,
              "curveFrom": "{0.9959328788762849, 0.9515271652805768}",
              "curveMode": 4,
              "curveTo": "{1.0061499834560077, 0.9717559860689977}",
              "hasCurveFrom": false,
              "hasCurveTo": true,
              "point": "{0.9959328788762849, 0.9515271652805768}",
            },
          ],
          "resizingConstraint": 63,
          "resizingType": 0,
          "rotation": 0,
          "shouldBreakMaskChain": false,
          "style": Object {
            "_class": "style",
            "borderOptions": Object {
              "_class": "borderOptions",
              "dashPattern": Array [],
              "isEnabled": true,
              "lineCapStyle": 0,
              "lineJoinStyle": 0,
            },
            "borders": Array [],
            "colorControls": Object {
              "_class": "colorControls",
              "brightness": 0,
              "contrast": 1,
              "hue": 0,
              "isEnabled": false,
              "saturation": 1,
            },
            "contextSettings": Object {
              "_class": "graphicsContextSettings",
              "blendMode": 0,
              "opacity": 1,
            },
            "do_objectID": "UUID",
            "endMarkerType": 0,
            "fills": Array [
              Object {
                "_class": "fill",
                "color": Object {
                  "_class": "color",
                  "alpha": 0.85,
                  "blue": 0,
                  "green": 0,
                  "red": 0,
                },
                "contextSettings": Object {
                  "_class": "graphicsContextSettings",
                  "blendMode": 0,
                  "opacity": 1,
                },
                "fillType": 0,
                "gradient": Object {
                  "_class": "gradient",
                  "elipseLength": 0,
                  "from": "{0, 0}",
                  "gradientType": 0,
                  "stops": Array [],
                  "to": "{1, 0}",
                },
                "isEnabled": true,
                "noiseIndex": 0,
                "noiseIntensity": 0,
                "patternFillType": 1,
                "patternTileScale": 1,
              },
            ],
            "innerShadows": Array [],
            "miterLimit": 10,
            "shadows": Array [],
            "startMarkerType": 0,
            "windingRule": 1,
          },
        },
        Object {
          "_class": "shapeGroup",
          "booleanOperation": -1,
          "clippingMaskMode": 0,
          "do_objectID": "UUID",
          "exportOptions": Object {
            "_class": "exportOptions",
            "exportFormats": Array [],
            "includedLayerIds": Array [],
            "layerOptions": 0,
            "shouldTrim": false,
          },
          "frame": Object {
            "_class": "rect",
            "constrainProportions": false,
            "height": 14,
            "width": 14,
            "x": 0,
            "y": 0,
          },
          "hasClickThrough": false,
          "hasClippingMask": false,
          "isFixedToViewport": false,
          "isFlippedHorizontal": false,
          "isFlippedVertical": false,
          "isLocked": false,
          "isVisible": true,
          "layerListExpandedType": 0,
          "layers": Array [
            Object {
              "_class": "shapePath",
              "booleanOperation": -1,
              "clippingMaskMode": 0,
              "do_objectID": "UUID",
              "edited": true,
              "exportOptions": Object {
                "_class": "exportOptions",
                "exportFormats": Array [],
                "includedLayerIds": Array [],
                "layerOptions": 0,
                "shouldTrim": false,
              },
              "frame": Object {
                "_class": "rect",
                "constrainProportions": false,
                "height": 14,
                "width": 14,
                "x": 0,
                "y": 0,
              },
              "isClosed": true,
              "isFixedToViewport": false,
              "isFlippedHorizontal": false,
              "isFlippedVertical": false,
              "isLocked": false,
              "isVisible": true,
              "layerListExpandedType": 0,
              "name": "路径",
              "nameIsFixed": false,
              "pointRadiusBehaviour": 1,
              "points": Array [
                Object {
                  "_class": "curvePoint",
                  "cornerRadius": 0,
                  "curveFrom": "{0.2238839285714286, 0}",
                  "curveMode": 3,
                  "curveTo": "{0.7761160714285714, 0}",
                  "hasCurveFrom": true,
                  "hasCurveTo": true,
                  "point": "{0.5, 0}",
                },
                Object {
                  "_class": "curvePoint",
                  "cornerRadius": 0,
                  "curveFrom": "{0, 0.7761160714285714}",
                  "curveMode": 3,
                  "curveTo": "{0, 0.2238839285714286}",
                  "hasCurveFrom": true,
                  "hasCurveTo": true,
                  "point": "{0, 0.5}",
                },
                Object {
                  "_class": "curvePoint",
                  "cornerRadius": 0,
                  "curveFrom": "{0.7761160714285714, 1}",
                  "curveMode": 3,
                  "curveTo": "{0.2238839285714286, 1}",
                  "hasCurveFrom": true,
                  "hasCurveTo": true,
                  "point": "{0.5, 1}",
                },
                Object {
                  "_class": "curvePoint",
                  "cornerRadius": 0,
                  "curveFrom": "{1, 0.2238839285714286}",
                  "curveMode": 3,
                  "curveTo": "{1, 0.7761160714285714}",
                  "hasCurveFrom": true,
                  "hasCurveTo": true,
                  "point": "{1, 0.5}",
                },
              ],
              "resizingConstraint": 63,
              "resizingType": 0,
              "rotation": 0,
              "shouldBreakMaskChain": false,
              "style": Object {
                "_class": "style",
                "borderOptions": Object {
                  "_class": "borderOptions",
                  "dashPattern": Array [],
                  "isEnabled": true,
                  "lineCapStyle": 0,
                  "lineJoinStyle": 0,
                },
                "borders": Array [],
                "colorControls": Object {
                  "_class": "colorControls",
                  "brightness": 0,
                  "contrast": 1,
                  "hue": 0,
                  "isEnabled": false,
                  "saturation": 1,
                },
                "contextSettings": Object {
                  "_class": "graphicsContextSettings",
                  "blendMode": 0,
                  "opacity": 1,
                },
                "do_objectID": "UUID",
                "endMarkerType": 0,
                "fills": Array [],
                "innerShadows": Array [],
                "miterLimit": 10,
                "shadows": Array [],
                "startMarkerType": 0,
                "windingRule": 1,
              },
            },
            Object {
              "_class": "shapePath",
              "booleanOperation": -1,
              "clippingMaskMode": 0,
              "do_objectID": "UUID",
              "edited": true,
              "exportOptions": Object {
                "_class": "exportOptions",
                "exportFormats": Array [],
                "includedLayerIds": Array [],
                "layerOptions": 0,
                "shouldTrim": false,
              },
              "frame": Object {
                "_class": "rect",
                "constrainProportions": false,
                "height": 11.625,
                "width": 11.625,
                "x": 1.1875,
                "y": 1.1875,
              },
              "isClosed": true,
              "isFixedToViewport": false,
              "isFlippedHorizontal": false,
              "isFlippedVertical": false,
              "isLocked": false,
              "isVisible": true,
              "layerListExpandedType": 0,
              "name": "路径",
              "nameIsFixed": false,
              "pointRadiusBehaviour": 1,
              "points": Array [
                Object {
                  "_class": "curvePoint",
                  "cornerRadius": 0,
                  "curveFrom": "{0.22392473118279574, 1}",
                  "curveMode": 3,
                  "curveTo": "{0.7760752688172042, 1}",
                  "hasCurveFrom": true,
                  "hasCurveTo": true,
                  "point": "{0.5, 1}",
                },
                Object {
                  "_class": "curvePoint",
                  "cornerRadius": 0,
                  "curveFrom": "{0, 0.22392473118279574}",
                  "curveMode": 3,
                  "curveTo": "{0, 0.7760752688172042}",
                  "hasCurveFrom": true,
                  "hasCurveTo": true,
                  "point": "{0, 0.5}",
                },
                Object {
                  "_class": "curvePoint",
                  "cornerRadius": 0,
                  "curveFrom": "{0.7760752688172042, 0}",
                  "curveMode": 3,
                  "curveTo": "{0.22392473118279574, 0}",
                  "hasCurveFrom": true,
                  "hasCurveTo": true,
                  "point": "{0.5, 0}",
                },
                Object {
                  "_class": "curvePoint",
                  "cornerRadius": 0,
                  "curveFrom": "{1, 0.7760752688172042}",
                  "curveMode": 3,
                  "curveTo": "{1, 0.22392473118279574}",
                  "hasCurveFrom": true,
                  "hasCurveTo": true,
                  "point": "{1, 0.5}",
                },
              ],
              "resizingConstraint": 63,
              "resizingType": 0,
              "rotation": 0,
              "shouldBreakMaskChain": false,
              "style": Object {
                "_class": "style",
                "borderOptions": Object {
                  "_class": "borderOptions",
                  "dashPattern": Array [],
                  "isEnabled": true,
                  "lineCapStyle": 0,
                  "lineJoinStyle": 0,
                },
                "borders": Array [],
                "colorControls": Object {
                  "_class": "colorControls",
                  "brightness": 0,
                  "contrast": 1,
                  "hue": 0,
                  "isEnabled": false,
                  "saturation": 1,
                },
                "contextSettings": Object {
                  "_class": "graphicsContextSettings",
                  "blendMode": 0,
                  "opacity": 1,
                },
                "do_objectID": "UUID",
                "endMarkerType": 0,
                "fills": Array [],
                "innerShadows": Array [],
                "miterLimit": 10,
                "shadows": Array [],
                "startMarkerType": 0,
                "windingRule": 1,
              },
            },
          ],
          "name": "shapeGroup",
          "nameIsFixed": false,
          "resizingConstraint": 63,
          "resizingType": 0,
          "rotation": 0,
          "shouldBreakMaskChain": false,
          "style": Object {
            "_class": "style",
            "borderOptions": Object {
              "_class": "borderOptions",
              "dashPattern": Array [],
              "isEnabled": true,
              "lineCapStyle": 0,
              "lineJoinStyle": 0,
            },
            "borders": Array [],
            "colorControls": Object {
              "_class": "colorControls",
              "brightness": 0,
              "contrast": 1,
              "hue": 0,
              "isEnabled": false,
              "saturation": 1,
            },
            "contextSettings": Object {
              "_class": "graphicsContextSettings",
              "blendMode": 0,
              "opacity": 1,
            },
            "do_objectID": "UUID",
            "endMarkerType": 0,
            "fills": Array [
              Object {
                "_class": "fill",
                "color": Object {
                  "_class": "color",
                  "alpha": 0.85,
                  "blue": 0,
                  "green": 0,
                  "red": 0,
                },
                "contextSettings": Object {
                  "_class": "graphicsContextSettings",
                  "blendMode": 0,
                  "opacity": 1,
                },
                "fillType": 0,
                "gradient": Object {
                  "_class": "gradient",
                  "elipseLength": 0,
                  "from": "{0, 0}",
                  "gradientType": 0,
                  "stops": Array [],
                  "to": "{1, 0}",
                },
                "isEnabled": true,
                "noiseIndex": 0,
                "noiseIntensity": 0,
                "patternFillType": 1,
                "patternTileScale": 1,
              },
            ],
            "innerShadows": Array [],
            "miterLimit": 10,
            "shadows": Array [],
            "startMarkerType": 0,
            "windingRule": 1,
          },
          "windingRule": 1,
        },
      ],
      "name": "up-circle",
      "nameIsFixed": false,
      "resizingConstraint": 63,
      "resizingType": 0,
      "rotation": 0,
      "shouldBreakMaskChain": false,
      "style": Object {
        "_class": "style",
        "borderOptions": Object {
          "_class": "borderOptions",
          "dashPattern": Array [],
          "isEnabled": true,
          "lineCapStyle": 0,
          "lineJoinStyle": 0,
        },
        "borders": Array [],
        "colorControls": Object {
          "_class": "colorControls",
          "brightness": 0,
          "contrast": 1,
          "hue": 0,
          "isEnabled": false,
          "saturation": 1,
        },
        "contextSettings": Object {
          "_class": "graphicsContextSettings",
          "blendMode": 0,
          "opacity": 1,
        },
        "do_objectID": "UUID",
        "endMarkerType": 0,
        "fills": Array [],
        "innerShadows": Array [],
        "miterLimit": 10,
        "shadows": Array [],
        "startMarkerType": 0,
        "windingRule": 1,
      },
    },
    Object {
      "_class": "group",
      "booleanOperation": -1,
      "clippingMaskMode": 0,
      "do_objectID": "UUID",
      "exportOptions": Object {
        "_class": "exportOptions",
        "exportFormats": Array [],
        "includedLayerIds": Array [],
        "layerOptions": 0,
        "shouldTrim": false,
      },
      "frame": Object {
        "_class": "rect",
        "constrainProportions": false,
        "height": 11.25,
        "width": 10.75,
        "x": 15.625,
        "y": 5.125,
      },
      "groupLayout": Object {
        "_class": "MSImmutableFreeformGroupLayout",
      },
      "hasClickThrough": false,
      "hasClippingMask": false,
      "isFixedToViewport": false,
      "isFlippedHorizontal": false,
      "isFlippedVertical": false,
      "isLocked": false,
      "isVisible": true,
      "layerListExpandedType": 0,
      "layers": Array [
        Object {
          "_class": "shapePath",
          "booleanOperation": -1,
          "clippingMaskMode": 0,
          "do_objectID": "UUID",
          "edited": true,
          "exportOptions": Object {
            "_class": "exportOptions",
            "exportFormats": Array [],
            "includedLayerIds": Array [],
            "layerOptions": 0,
            "shouldTrim": false,
          },
          "frame": Object {
            "_class": "rect",
            "constrainProportions": false,
            "height": 11.25,
            "width": 1.1875,
            "x": 4.78125,
            "y": 4.440892098500626e-16,
          },
          "isClosed": true,
          "isFixedToViewport": false,
          "isFlippedHorizontal": false,
          "isFlippedVertical": false,
          "isLocked": false,
          "isVisible": true,
          "layerListExpandedType": 0,
          "name": "路径",
          "nameIsFixed": false,
          "pointRadiusBehaviour": 1,
          "points": Array [
            Object {
              "_class": "curvePoint",
              "cornerRadius": 0,
              "curveFrom": "{0.10526315789473684, 0}",
              "curveMode": 4,
              "curveTo": "{0.03508771929824561, 0}",
              "hasCurveFrom": false,
              "hasCurveTo": true,
              "point": "{0.10526315789473684, 0}",
            },
            Object {
              "_class": "curvePoint",
              "cornerRadius": 0,
              "curveFrom": "{0.9649122807017543, 0}",
              "curveMode": 4,
              "curveTo": "{0.8947368421052632, 0}",
              "hasCurveFrom": true,
              "hasCurveTo": false,
              "point": "{0.8947368421052632, 0}",
            },
            Object {
              "_class": "curvePoint",
              "cornerRadius": 0,
              "curveFrom": "{1, 0.011111111111111112}",
              "curveMode": 4,
              "curveTo": "{1, 0.0037037037037037034}",
              "hasCurveFrom": false,
              "hasCurveTo": true,
              "point": "{1, 0.011111111111111112}",
            },
            Object {
              "_class": "curvePoint",
              "cornerRadius": 0,
              "curveFrom": "{1, 0.9962962962962963}",
              "curveMode": 4,
              "curveTo": "{1, 0.9888888888888889}",
              "hasCurveFrom": true,
              "hasCurveTo": false,
              "point": "{1, 0.9888888888888889}",
            },
            Object {
              "_class": "curvePoint",
              "cornerRadius": 0,
              "curveFrom": "{0.8947368421052632, 1}",
              "curveMode": 4,
              "curveTo": "{0.9649122807017543, 1}",
              "hasCurveFrom": false,
              "hasCurveTo": true,
              "point": "{0.8947368421052632, 1}",
            },
            Object {
              "_class": "curvePoint",
              "cornerRadius": 0,
              "curveFrom": "{0.03508771929824561, 1}",
              "curveMode": 4,
              "curveTo": "{0.10526315789473684, 1}",
              "hasCurveFrom": true,
              "hasCurveTo": false,
              "point": "{0.10526315789473684, 1}",
            },
            Object {
              "_class": "curvePoint",
              "cornerRadius": 0,
              "curveFrom": "{0, 0.9888888888888889}",
              "curveMode": 4,
              "curveTo": "{0, 0.9962962962962963}",
              "hasCurveFrom": false,
              "hasCurveTo": true,
              "point": "{0, 0.9888888888888889}",
            },
            Object {
              "_class": "curvePoint",
              "cornerRadius": 0,
              "curveFrom": "{0, 0.0037037037037037034}",
              "curveMode": 4,
              "curveTo": "{0, 0.011111111111111112}",
              "hasCurveFrom": true,
              "hasCurveTo": false,
              "point": "{0, 0.011111111111111112}",
            },
          ],
          "resizingConstraint": 63,
          "resizingType": 0,
          "rotation": 0,
          "shouldBreakMaskChain": false,
          "style": Object {
            "_class": "style",
            "borderOptions": Object {
              "_class": "borderOptions",
              "dashPattern": Array [],
              "isEnabled": true,
              "lineCapStyle": 0,
              "lineJoinStyle": 0,
            },
            "borders": Array [],
            "colorControls": Object {
              "_class": "colorControls",
              "brightness": 0,
              "contrast": 1,
              "hue": 0,
              "isEnabled": false,
              "saturation": 1,
            },
            "contextSettings": Object {
              "_class": "graphicsContextSettings",
              "blendMode": 0,
              "opacity": 1,
            },
            "do_objectID": "UUID",
            "endMarkerType": 0,
            "fills": Array [
              Object {
                "_class": "fill",
                "color": Object {
                  "_class": "color",
                  "alpha": 0.85,
                  "blue": 0,
                  "green": 0,
                  "red": 0,
                },
                "contextSettings": Object {
                  "_class": "graphicsContextSettings",
                  "blendMode": 0,
                  "opacity": 1,
                },
                "fillType": 0,
                "gradient": Object {
                  "_class": "gradient",
                  "elipseLength": 0,
                  "from": "{0, 0}",
                  "gradientType": 0,
                  "stops": Array [],
                  "to": "{1, 0}",
                },
                "isEnabled": true,
                "noiseIndex": 0,
                "noiseIntensity": 0,
                "patternFillType": 1,
                "patternTileScale": 1,
              },
            ],
            "innerShadows": Array [],
            "miterLimit": 10,
            "shadows": Array [],
            "startMarkerType": 0,
            "windingRule": 1,
          },
        },
        Object {
          "_class": "shapePath",
          "booleanOperation": -1,
          "clippingMaskMode": 0,
          "do_objectID": "UUID",
          "edited": true,
          "exportOptions": Object {
            "_class": "exportOptions",
            "exportFormats": Array [],
            "includedLayerIds": Array [],
            "layerOptions": 0,
            "shouldTrim": false,
          },
          "frame": Object {
            "_class": "rect",
            "constrainProportions": false,
            "height": 1.1875,
            "width": 10.75,
            "x": 0,
            "y": 5.03125,
          },
          "isClosed": true,
          "isFixedToViewport": false,
          "isFlippedHorizontal": false,
          "isFlippedVertical": false,
          "isLocked": false,
          "isVisible": true,
          "layerListExpandedType": 0,
          "name": "路径",
          "nameIsFixed": false,
          "pointRadiusBehaviour": 1,
          "points": Array [
            Object {
              "_class": "curvePoint",
              "cornerRadius": 0,
              "curveFrom": "{0.011627906976744186, 0}",
              "curveMode": 4,
              "curveTo": "{0.003875968992248062, 0}",
              "hasCurveFrom": false,
              "hasCurveTo": true,
              "point": "{0.011627906976744186, 0}",
            },
            Object {
              "_class": "curvePoint",
              "cornerRadius": 0,
              "curveFrom": "{0.996124031007752, 0}",
              "curveMode": 4,
              "curveTo": "{0.9883720930232558, 0}",
              "hasCurveFrom": true,
              "hasCurveTo": false,
              "point": "{0.9883720930232558, 0}",
            },
            Object {
              "_class": "curvePoint",
              "cornerRadius": 0,
              "curveFrom": "{1, 0.10526315789473684}",
              "curveMode": 4,
              "curveTo": "{1, 0.03508771929824561}",
              "hasCurveFrom": false,
              "hasCurveTo": true,
              "point": "{1, 0.10526315789473684}",
            },
            Object {
              "_class": "curvePoint",
              "cornerRadius": 0,
              "curveFrom": "{1, 0.9649122807017543}",
              "curveMode": 4,
              "curveTo": "{1, 0.8947368421052632}",
              "hasCurveFrom": true,
              "hasCurveTo": false,
              "point": "{1, 0.8947368421052632}",
            },
            Object {
              "_class": "curvePoint",
              "cornerRadius": 0,
              "curveFrom": "{0.9883720930232558, 1}",
              "curveMode": 4,
              "curveTo": "{0.996124031007752, 1}",
              "hasCurveFrom": false,
              "hasCurveTo": true,
              "point": "{0.9883720930232558, 1}",
            },
            Object {
              "_class": "curvePoint",
              "cornerRadius": 0,
              "curveFrom": "{0.003875968992248062, 1}",
              "curveMode": 4,
              "curveTo": "{0.011627906976744186, 1}",
              "hasCurveFrom": true,
              "hasCurveTo": false,
              "point": "{0.011627906976744186, 1}",
            },
            Object {
              "_class": "curvePoint",
              "cornerRadius": 0,
              "curveFrom": "{0, 0.8947368421052632}",
              "curveMode": 4,
              "curveTo": "{0, 0.9649122807017543}",
              "hasCurveFrom": false,
              "hasCurveTo": true,
              "point": "{0, 0.8947368421052632}",
            },
            Object {
              "_class": "curvePoint",
              "cornerRadius": 0,
              "curveFrom": "{0, 0.03508771929824561}",
              "curveMode": 4,
              "curveTo": "{0, 0.10526315789473684}",
              "hasCurveFrom": true,
              "hasCurveTo": false,
              "point": "{0, 0.10526315789473684}",
            },
          ],
          "resizingConstraint": 63,
          "resizingType": 0,
          "rotation": 0,
          "shouldBreakMaskChain": false,
          "style": Object {
            "_class": "style",
            "borderOptions": Object {
              "_class": "borderOptions",
              "dashPattern": Array [],
              "isEnabled": true,
              "lineCapStyle": 0,
              "lineJoinStyle": 0,
            },
            "borders": Array [],
            "colorControls": Object {
              "_class": "colorControls",
              "brightness": 0,
              "contrast": 1,
              "hue": 0,
              "isEnabled": false,
              "saturation": 1,
            },
            "contextSettings": Object {
              "_class": "graphicsContextSettings",
              "blendMode": 0,
              "opacity": 1,
            },
            "do_objectID": "UUID",
            "endMarkerType": 0,
            "fills": Array [
              Object {
                "_class": "fill",
                "color": Object {
                  "_class": "color",
                  "alpha": 0.85,
                  "blue": 0,
                  "green": 0,
                  "red": 0,
                },
                "contextSettings": Object {
                  "_class": "graphicsContextSettings",
                  "blendMode": 0,
                  "opacity": 1,
                },
                "fillType": 0,
                "gradient": Object {
                  "_class": "gradient",
                  "elipseLength": 0,
                  "from": "{0, 0}",
                  "gradientType": 0,
                  "stops": Array [],
                  "to": "{1, 0}",
                },
                "isEnabled": true,
                "noiseIndex": 0,
                "noiseIntensity": 0,
                "patternFillType": 1,
                "patternTileScale": 1,
              },
            ],
            "innerShadows": Array [],
            "miterLimit": 10,
            "shadows": Array [],
            "startMarkerType": 0,
            "windingRule": 1,
          },
        },
      ],
      "name": "plus",
      "nameIsFixed": false,
      "resizingConstraint": 63,
      "resizingType": 0,
      "rotation": 0,
      "shouldBreakMaskChain": false,
      "style": Object {
        "_class": "style",
        "borderOptions": Object {
          "_class": "borderOptions",
          "dashPattern": Array [],
          "isEnabled": true,
          "lineCapStyle": 0,
          "lineJoinStyle": 0,
        },
        "borders": Array [],
        "colorControls": Object {
          "_class": "colorControls",
          "brightness": 0,
          "contrast": 1,
          "hue": 0,
          "isEnabled": false,
          "saturation": 1,
        },
        "contextSettings": Object {
          "_class": "graphicsContextSettings",
          "blendMode": 0,
          "opacity": 1,
        },
        "do_objectID": "UUID",
        "endMarkerType": 0,
        "fills": Array [],
        "innerShadows": Array [],
        "miterLimit": 10,
        "shadows": Array [],
        "startMarkerType": 0,
        "windingRule": 1,
      },
    },
  ],
  "name": "编组",
  "nameIsFixed": false,
  "overrideProperties": Array [],
  "resizesContent": false,
  "resizingConstraint": 1,
  "resizingType": 1,
  "rotation": 0,
  "sharedStyleID": "",
  "shouldBreakMaskChain": false,
  "symbolID": "UUID",
  "verticalRulerData": Object {
    "_class": "rulerData",
    "base": 0,
    "guides": Array [],
  },
}
`;
